rs/nns/governance/src/governance.rs

```diff
@@ -6781,13 +6786,50 @@ impl Governance {
                     );
                 }
             }
-            self.schedule_pending_rewards_distribution(day_after_genesis, reward_distribution);
+            Some(reward_distribution)
+        };
+
+        let reward_event = RewardEvent {
+            day_after_genesis,
+            actual_timestamp_seconds: now,
+            settled_proposals: considered_proposals,
+            distributed_e8s_equivalent: actually_distributed_e8s_equivalent,
+            total_available_e8s_equivalent: total_available_e8s_equivalent_float as u64,
+            rounds_since_last_distribution: Some(rounds_since_last_distribution),
+            latest_round_available_e8s_equivalent: Some(
+                latest_round_available_e8s_equivalent_float as u64,
+            ),
+        };
+
+        (Some(reward_event), reward_distribution)
+    }
+
+    /// Create a reward event.
+    ///
+    /// This method:
+    /// * collects all proposals in state ReadyToSettle, that is, proposals that
+    ///   can no longer accept votes for the purpose of rewards and that have
+    ///   not yet been considered in a reward event.
+    /// * Associate those proposals to the new reward event
+    pub(crate) fn distribute_rewards(&mut self, supply: Tokens) {
+        println!("{}distribute_rewards. Supply: {:?}", LOG_PREFIX, supply);
+
+        let (new_reward_event, reward_distribution) = self.calculate_voting_rewards(supply);
```

Comment:
If (None, Some) is invalid (which I think it is), should there be any sort of debug_assert here?  Or some way of making sure such a condition is caught in a test?